### PR TITLE
Adds variables to the analytics_api master config YAML file

### PR DESCRIPTION
**Pull request by pomegranited**
Adds the following variables to the generated `/edx/etc/analytics_api.yml` file, allowing them to be configurable during provisioning:
- `DATE_FORMAT` and `DATETIME_FORMAT`: used to format dates and timestamps returned by the Analytics REST API.
- `DEFAULT_PAGE_SIZE` and `MAX_PAGE_SIZE`: used to paginate (currently only) Learner List API results.  However, it's likely that other paginated API results will want to use them too.

This change helps facility the use of the Analytics Learner API to generate downloadable reports tailored to customer needs.

**Discussions**: This change is related to [edx-analytics-data-api PR #128](https://github.com/edx/edx-analytics-data-api/pull/128), which has been discussed extensively with the Analytics team.

**Dependencies**: None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Partner information**: 3rd party-hosted open edX instance

**Testing instructions**:
1. See [edx-analytics-data-api PR #128](https://github.com/edx/edx-analytics-data-api/pull/128) for testing setup instructions.
2. Checkout this PR branch on your vagrant instance under `/edx/app/edx_ansible`
3. Create an extra-vars.yml file containing some overrides for the variables in this PR, and any others you want to preserve, e.g.
   
   ``` yaml
   ANALYTICS_API_DATE_FORMAT: "date goes here: %Y-%m-%d"
   ANALYTICS_API_DATETIME_FORMAT: iso-8601
   ANALYTICS_API_DEFAULT_PAGE_SIZE: 1
   ANALYTICS_API_MAX_PAGE_SIZE: 3
   
   ANALYTICS_API_ELASTICSEARCH_LEARNERS_HOST: localhost:9223
   ANALYTICS_API_ELASTICSEARCH_LEARNERS_INDEX: roster
   ```
4. Re-run the updated `analytics_api` role:
   
   ```
   ssh vagrant
   source /edx/app/edx_ansible/venvs/edx_ansible/bin/activate
   cd /edx/app/edx_ansible/edx_ansible/playbooks
   ansible-playbook -i localhost, -c local run_role.yml -e role=analytics_api --extra-vars=@/path/to/extra-vars.yml
   ```
5. Note changes written to `/edx/etc/analytics_api.yml`.
6. Confirm the changes using the [Learner List API](http://localhost:8100/docs/#!/api/Learner_List):
   - Check `DATE*_FORMAT` changes by looking at the `last_updated` date field.
   - Check the `*_PAGE_SIZE` changes counting the number of returned records.

**Reviewers**
- [x] @haikuginger
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fconfiguration&number=3273

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @pomegranited! I've created [OSPR-1361](https://openedx.atlassian.net/browse/OSPR-1361) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by haikuginger** [comment]
:+1:
- [x] I tested this: I verified that the Ansible playbook set the appropriate variables in `/edx/etc/analytics-api.yml`, and once those variables were in place, I verified that they altered the behavior of the analytics API.
- [x] I read through the code
- [x] No additional documentation necesssary

**Comment by pomegranited** [comment]
@mulby CC @edx/devops This issue is ready for review, whenever you get a chance.

**Comment by feanil** [comment]
:+1:   I verified that the defaults match the ones in the analytics repo.  @mulby please feel free to merge this once you're happy with it.

File: playbooks/roles/analytics_api/defaults/main.yml [hunk 1]
```diff
@@ -62,6 +62,10 @@ ANALYTICS_API_ELASTICSEARCH_CONNECTION_CLASS: !!null
 ANALYTICS_API_ELASTICSEARCH_AWS_ACCESS_KEY_ID: !!null
 ANALYTICS_API_ELASTICSEARCH_AWS_SECRET_ACCESS_KEY: !!null
 ANALYTICS_API_ELASTICSEARCH_CONNECTION_DEFAULT_REGION: 'us-east-1'
+ANALYTICS_API_DATE_FORMAT: '%Y-%m-%d'
+ANALYTICS_API_DATETIME_FORMAT: '%Y-%m-%dT%H%M%S'
+ANALYTICS_API_DEFAULT_PAGE_SIZE: 10
```
Comment (mulby) [inline playbooks/roles/analytics_api/defaults/main.yml]: This is currently 25, isn't it?

https://github.com/edx/edx-analytics-data-api/pull/128/files#diff-d3a1839d19542dd649d35dedd6ca0b5fL1

File: playbooks/roles/analytics_api/defaults/main.yml [hunk 2]
```diff
@@ -62,6 +62,10 @@ ANALYTICS_API_ELASTICSEARCH_CONNECTION_CLASS: !!null
...
+ANALYTICS_API_MAX_PAGE_SIZE: 25
```
Comment (mulby) [inline playbooks/roles/analytics_api/defaults/main.yml]: And this is currently 100?

https://github.com/edx/edx-analytics-data-api/pull/128/files#diff-3211e1311bf0cc9152631fd015a4bab1L193

Comment (pomegranited) [inline playbooks/roles/analytics_api/defaults/main.yml]: @mulby  Oh gosh, sorry about that!  Fixed with [899ec48](https://github.com/edx/configuration/pull/3273/commits/899ec486b6aaedeebc02ac64fc7081624e9b7d26).

**Pull request merged**